src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.dao.UserDao;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        return userDao.add(user);
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        Optional<User> getUserFromDB = userDao.findByEmail(email);
+        if (getUserFromDB.isPresent()) {
+            return getUserFromDB;
+        }
+
+        return null;
```

Comment:
what the point of making check and return optional anyway?

Reply:
I accidentally did that, I changed it back